catalog/ui/search-ui/standard/src/main/webapp/js/model/user.js

```diff
@@ -31,47 +32,55 @@ define([
         geometryCollectionColor: '#FFFF67'
     };
 
-    var mapLayerConfigForURL = {};
-
-    _.each(properties.imageryProviders, function (providerConfig, index) {
-        /*
-         - prefer configured 'index' value
-         - fallback to implicit order in admin ui.
-         - duplicate indexes sorted arbitrarily.
-         - allow semantic values like -1000 ~ "lowest", 1000 ~ "highest"
-         */
-        var configIndex;
-        if (providerConfig.index && !isNaN(Number(providerConfig.index))) {
-            configIndex = Number(providerConfig.index);
-        }
-        else {
-            configIndex = index;
+    function generateId() {
+        return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function (c) {
+            var r = Math.random() * 16 | 0,
+                v = c === 'x' ? r : (r & 0x3 | 0x8);
+            return v.toString(16);
+        });
+    }
+
+    User.updateMapLayers = function(layerPrefs, layersToRemove, index) {
+        layerPrefs.each(function (layer) {
+            var found = false;
+            if (layer) {
+                for (var i = 0; i < properties.imageryProviders.length; i++) {
```

Comment:
You can use forEach here.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach